buildpacks/nodejs-npm-install/src/errors.rs

```diff
@@ -0,0 +1,169 @@
+use crate::npm;
+use commons::fun_run::CmdError;
+use commons::output::build_log::{BuildLog, Logger, StartedLogger};
+use commons::output::fmt::DEBUG_INFO;
+use heroku_nodejs_utils::application;
+use heroku_nodejs_utils::package_json::PackageJsonError;
+use indoc::formatdoc;
+use std::fmt::Display;
+use std::io::stdout;
+
+const OPEN_A_SUPPORT_TICKET: &str =
+    "open a support ticket and include the full log output of this build";
+
+const TRY_BUILDING_AGAIN: &str = "try again and to see if the error resolves itself";
+
+#[derive(Debug)]
+pub(crate) enum NpmInstallBuildpackError {
+    Application(application::Error),
+    BuildScript(CmdError),
+    NpmInstall(CmdError),
+    NpmSetCacheDir(CmdError),
+    NpmVersion(npm::VersionError),
+    PackageJson(PackageJsonError),
+}
+
+pub(crate) fn on_error(error: libcnb::Error<NpmInstallBuildpackError>) {
+    let logger = BuildLog::new(stdout()).without_buildpack_name();
+    match error {
+        libcnb::Error::BuildpackError(buildpack_error) => {
+            on_buildpack_error(buildpack_error, logger)
+        }
+        framework_error => on_framework_error(framework_error, logger),
+    }
+}
+
+pub(crate) fn on_buildpack_error(error: NpmInstallBuildpackError, logger: Box<dyn StartedLogger>) {
+    match error {
+        NpmInstallBuildpackError::PackageJson(e) => on_package_json_error(e, logger),
+        NpmInstallBuildpackError::NpmSetCacheDir(e) => on_set_cache_dir_error(e, logger),
+        NpmInstallBuildpackError::NpmVersion(e) => on_npm_version_error(e, logger),
+        NpmInstallBuildpackError::NpmInstall(e) => on_npm_install_error(e, logger),
+        NpmInstallBuildpackError::BuildScript(e) => on_build_script_error(e, logger),
+        NpmInstallBuildpackError::Application(e) => on_application_error(e, logger),
+    }
+}
+
+fn on_package_json_error(error: PackageJsonError, logger: Box<dyn StartedLogger>) {
+    match error {
+        PackageJsonError::AccessError(e) => {
+            print_error_details(logger, e)
+                .announce()
+                .error(&formatdoc! {"
+                    Failed to read package.json
+
+                    An unexpected error occurred while reading package.json. 
+                    
+                    Please {TRY_BUILDING_AGAIN}. 
+                "});
+        }
+        PackageJsonError::ParseError(e) => {
+            print_error_details(logger, e)
+                .announce()
+                .error(&formatdoc! {"
+                    Failed to parse package.json
+
+                    An unexpected error occurred while parsing package.json. 
+                    
+                    Please {TRY_BUILDING_AGAIN}.   
```

Comment:
If we failed to parse the `package.json`, building again is unlikely to help. Maybe the suggestion should be to check the formatting of the file?